library/common.h

```diff
@@ -390,6 +391,32 @@ extern void (*mbedtls_test_hook_test_fail)( const char * test, int line, const c
 }
 #endif
 
+/**
+ * Perform a fast block XOR operation, such that
+ * r[i] = a[i] ^ b[i] where 0 <= i < n
+ *
+ * \param   r Pointer to result (buffer of at least \p n bytes). \p r
+ *            may be equal to either \p a or \p b, but behaviour when
+ *            it overlaps in other ways is undefined.
+ * \param   a Pointer to input (buffer of at least \p n bytes)
+ * \param   b Pointer to input (buffer of at least \p n bytes)
+ * \param   n Number of bytes to process.
+ */
+static inline void mbedtls_xor( unsigned char* r, unsigned char const *a, unsigned char const *b, size_t n )
```

Comment:
This is really big for a `static inline` function, and will be even worse with support for unaligned accesses. Please move the definition to a .c file.

A [C99](https://en.cppreference.com/w/c/language/inline) `static inline` definition in a header with an `extern inline` this-declaration-is-actually-the-definition indicator would be ideal. We aren't using `extern inline` yet, but we've officially assumed C99 for several years now, so we can retire any compiler that doesn't support that. We may have to fiddle with how we invoke GCC on our CI since older GCC had the same feature but with a different syntax.

Reply:
If the definition is in a separate C file then surely it won't get inlined without LTO enabled (except within the C file where it is defined)? I've tested without LTO and can't see any inlining.

Reply:
The C99 way is:

* `inline foo() { … }` in a header file. (My bad, I wrote `static inline` earlier.)
* `extern inline foo();` in a .c file. (Yes, you have to write `extern` to mean “this is where to put the object”, which is exactly the opposite of what `extern` means elsewhere.)

Then there's a linkable copy in the .o file, but the compiler has the definition available if it chooses to inline and optimize.

Reply:
OK, thanks, that makes sense. I experimented and I wasn't able to get clang to not inline it, even with `-fno-inline-functions -Oz` so the result was, ironically, a slightly larger `libmbedcrypto.a` (presumably because it contains the extra, unused copy) - but LTO would remove it again I suppose.

Reply:
I kind of agree about `platform_util.c` not being quite right, but also agree there's no clear better option. For now I think `platform_util.c` is probably best choice.